guardian/shortcuts.py

```diff
@@ -393,8 +396,9 @@ def get_objects_for_user(user, perms, klass=None, use_groups=True, any_perm=Fals
     user_model = get_user_obj_perms_model(queryset.model)
     user_obj_perms_queryset = (user_model.objects
         .filter(user=user)
-        .filter(permission__content_type=ctype)
-        .filter(permission__codename__in=codenames))
+        .filter(permission__content_type=ctype))
+    if len(codenames):
+        user_obj_perms_queryset = user_obj_perms_queryset.filter(permission__codename__in=codenames)
```

Comment:
See here. Instead of just making it always filter by `codenames`, it filters only when it is non-zero (so  when `ctype` was specified). This allows the behavior of getting any objects for any permission.